### DISCUSSION TITLE
Anyone used with Drupal

### DISCUSSION THREAD

**Discussion by stubar** [nextauthjs/next-auth#3680] [Help]
I'm trying to login to a Drupal OAuth server https://www.drupal.org/project/oauth2_server, which is built upon https://github.com/bshaffer/oauth2-server-php.

We kind of got it working in next-auth 3.1 but it was very flakey and we really need to upgrade, so I am trying to use it with v4.1.2.

I am setting a single custom provider  in the providers array that looks like this 
```
{
      id: 'THE',
      name: 'THE',
      type: 'oauth',
      version: '2.0',
      token: `${UMSServerUrl}/oauth2/token`,
      userinfo: `${UMSServerUrl}/oauth2/UserInfo`,
      authorization: {
        params: { scope: 'openid email profile' },
        url: `${UMSServerUrl}/oauth2/authorize?response_type=code`
      },

      profile: (profile: any) => ({
        id: profile.sub,
        sub: profile.sub,
        email: profile.email,
        email_verified: profile.email_verified,
        name: profile.name,
        preferred_username: profile.preferred_username,
        zoneinfo: profile.zoneinfo
      }),
      clientId: 'ums_idp',
      clientSecret,
      checks: 'state',
      idToken: true,
      issuer: `${UMSServerUrl}/`
    }
```
I get an error from the openid client `error: Object, providerId: "THE", message: "jwks_uri must be configured on the issuer"`

This is from the oAuthCallback method.  I've tried tweaking numerous settings in my provider and some of the other options but I'm shooting in the dark really.  Has anyone got any experience of connecting to this oAuth server?

**Comment by PBnJson** [comment]
Hey, just wondering if you ever got this working the way you want?? I have a custom provider as well I'm trying to use with Drupal and Classlink.

**Reply by stubar** [reply]
I have left that company and never managed to upgrade in my time there I'm afraid.
